src/MvvmNano.Forms/MvvmNanoFormsPresenter.cs

```diff
@@ -119,7 +132,16 @@ public void SetApplication(Application application)
             IView view = CreateViewFor<TViewModel>();
             view.SetViewModel(viewModel);
 
-            return OpenPageAsync(view as Page);
+            return StartOpeningPageAsync<TViewModel>(view as Page);
+        }
+
+        /// <summary>
+        /// Opens the page a <see cref="MasterDetailData.ViewModelType"/> is referencing to.
+        /// </summary>
+        /// <param name="data"></param>
+        public void SetDetail(MasterDetailData data)
```

Comment:
This one `SetDetail()` and `GetViewNameByViewModel()` are public now and not part of the contract of `IPresenter`. First thought: That should be changed. Second thought: Wait ... 

1. `MvvmNanoMasterDetailPage.MenuEntrySelected()` calls `_presenter.SetDetail()`
2. From here we hack through to `NavigateToViewModel()` which ends up in `StartOpeningPageAsync()`
3. Which then, at the end of a long day, is calling `MvvmNanoMasterDetailPage.SetDetail()` again.

Why not leaving the presenter out of that specific game and doing all this stuff in `MvvmNanoMasterDetailPage` itself.

Von hinten die Brust ins Auge ;-)